ui/src/main/webapp/src/app/project/migration-project.service.ts

```diff
@@ -132,6 +139,20 @@ export class MigrationProjectService extends AbstractService {
             .catch(this.handleError);
     }
 
+    @Cached('project', {minutes: 1})
+    getAllIncludingProvisional(): Observable<Array<MigrationProject & HasAppCount>> {
+        return this._http.get(Constants.REST_BASE + this.GET_MIGRATION_PROJECTS_URL + "?includeProvisional=true")
+            .map(res => <MigrationProjectAndCount[]> res.json())
+            .map(entries => entries.map(entry => (entry.migrationProject["applicationCount"] = entry.applicationCount, entry.migrationProject)))
+            .catch(this.handleError);
+    }
+
+    deleteProvisionalProjects(): Observable<void> {
+        return this._http.delete(Constants.REST_BASE + this.DELETE_PROVISIONAL_PROJECTS_URL)
```

Comment:
Maybe I should repeat the explanation here.
The thing is, you end up needing to do the action when the create dialog is opened. Doing it afterwards, where you could do it as a part of some `createProject()` doesn't make much sense. And at that point, there's actually no call eligible for binding this cleanup to. Hence an extra endpoint.